### PR TITLE
Playlist thumbnail

### PR DESCRIPTION
**Pull request by hig-dev**
This PR adds a playlist thumbnail to the playlist metadata.
The playlist thumbnail is from the first video of a playlist.

### CODE REVIEW

File: YoutubeExplode/ReverseEngineering/Responses/PlaylistResponse.cs [hunk 1]
```diff
@@ -28,6 +29,12 @@ internal partial class PlaylistResponse
             .GetPropertyOrNull("description")?
             .GetString();
 
+        public ThumbnailSet? TryGetThumbnails()
+        {
+            string? firstVideoId = GetVideos().FirstOrDefault()?.GetId();
+            return firstVideoId != null ? new ThumbnailSet(firstVideoId) : null;
+        }
```
Comment (Tyrrrz) [inline YoutubeExplode/ReverseEngineering/Responses/PlaylistResponse.cs:36]: Can rewrite this as:

```csharp
public ThumbnailSet? TryGetThumbnails() => GetVideos()
    .FirstOrDefault()?
    .GetId()
    .Pipe(i => new ThumbnailSet(i));

```

File: YoutubeExplode.Tests/PlaylistSpecs.cs [hunk 2]
```diff
@@ -23,6 +23,11 @@ public async Task I_can_get_metadata_of_a_YouTube_playlist()
             playlist.Title.Should().Be("osu! Highlights");
             playlist.Author.Should().Be("Tyrrrz");
             playlist.Description.Should().Be("My best osu! plays");
+            playlist.Thumbnails.LowResUrl.Should().NotBeNullOrWhiteSpace();
+            playlist.Thumbnails.MediumResUrl.Should().NotBeNullOrWhiteSpace();
+            playlist.Thumbnails.HighResUrl.Should().NotBeNullOrWhiteSpace();
+            playlist.Thumbnails.StandardResUrl.Should().NotBeNullOrWhiteSpace();
+            playlist.Thumbnails.MaxResUrl.Should().NotBeNullOrWhiteSpace();
```
Comment (Tyrrrz) [inline YoutubeExplode.Tests/PlaylistSpecs.cs:30]: `Thumbnails?.LowResUrl` to avoid null reference exception in case of a bug.
Same with others.

Comment (hig-dev) [reply YoutubeExplode.Tests/PlaylistSpecs.cs]: Done

File: YoutubeExplode/ReverseEngineering/Responses/PlaylistResponse.cs (hunk 1 above)
Comment (hig-dev) [reply YoutubeExplode/ReverseEngineering/Responses/PlaylistResponse.cs]: Done

**Comment by Tyrrrz** [comment]
Thanks! 🎉

**Pull request merged**